PrefectHQ/prefect
# Fix deprecation warnings in Python 3.10

## Summary

Fix deprecation warnings in Python 3.10

## Changes

* Use `logger.warning` instead of deprecated `logger.warn`.

```python
python -Wall
Python 3.10.3 (main, Mar 16 2022, 16:15:16) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> logger = logging.getLogger()
>>> logger.warn("test")
<stdin>:1: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
test
>>> logger.warning("test")
test
```

* Use `packaging.version` instead of deprecated `distutils`.

```python
tests/utilities/test_agent.py: 10 warnings
  /home/karthikeyan/stuff/python/prefect/src/prefect/utilities/agent.py:68: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(core_version) < LooseVersion("0.13.0"):
```

* Use `daemon` attribute instead of deprecated `isDaemon` method.

```python
tests/utilities/test_executors.py::test_heartbeat_is_daemonic_by_default
  /home/karthikeyan/stuff/python/prefect/tests/utilities/test_executors.py:388: DeprecationWarning: isDaemon() is deprecated, get the daemon attribute instead
    assert heartbeat.isDaemon()
```

* Use `get_running_loop` instead instead of `get_event_loop` that emits Deprecation warning when there is no running loop.

```python
tests/agent/test_agent.py::test_agent_poke_api
  /home/karthikeyan/stuff/python/prefect/src/prefect/agent/agent.py:464: DeprecationWarning: There is no current event loop
    asyncio.get_event_loop()
```

* Fix deprecation warning due to invalid escape sequences.

```python
find tests -iname '*py' | xargs -I{} python -Wall -m py_compile {} 
tests/tasks/mysql/test_mysql.py:44: DeprecationWarning: invalid escape sequence '\('
  match="The 'fetch' parameter must be one of the following - \('one', 'many', 'all'\)",
tests/tasks/mysql/test_mysql.py:74: DeprecationWarning: invalid escape sequence '\['
  match=f"'cursor_type' should be one of \['cursor', 'dictcursor', 'sscursor', 'ssdictcursor'\] or a full cursor class, got {cursor_type}",
tests/tasks/mysql/test_mysql.py:82: DeprecationWarning: invalid escape sequence '\['
  match=f"'cursor_type' should be one of \['cursor', 'dictcursor', 'sscursor', 'ssdictcursor'\] or a full cursor class, got \['cursor'\]",
(.py310env) ➜  prefect git:(master) ✗ pyupgrade tests/tasks/mysql/test_mysql.py
```



## Importance





## Checklist


This PR:

- [ ] adds new tests (if appropriate)
- [ ] adds a change file in the `changes/` directory (if appropriate)
- [ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)
